src/main/java/com/sonyericsson/jenkins/plugins/bfa/PluginImpl.java

```diff
@@ -239,6 +235,76 @@ public ScanOnDemandVariables getSodVariables() {
         return sodVariables;
     }
 
+    /**
+     * Used by jelly for loading the view, it can't seem to handle retrieving from the nested field.
+     *
+     * @return minimum sod worker threads
+     */
+    @Restricted(NoExternalUse.class)
+    public int getMinimumSodWorkerThreads() {
+        if (sodVariables == null) {
+            return ScanOnDemandVariables.DEFAULT_MINIMUM_SOD_WORKER_THREADS;
+        }
+
+        return sodVariables.getMinimumSodWorkerThreads();
```

Comment:
Why did you had to make the change in `ScanOnDemandBaseActionTest` if the potential NPE is handled here?

Reply:
ScanOnDemandVariables aren't set by default in the current setup, probably needs a readResolve fix to initialise them if they aren't set. I'll take a look

Reply:
The test didn't call these getters they were only called by jelly.
I've removed these getters now anyway.